sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -2687,7 +2687,7 @@ object DatePart {
       errorHandleFunc: => Nothing): Expression = extractField.toUpperCase(Locale.ROOT) match {
     case "YEAR" | "Y" | "YEARS" | "YR" | "YRS" => Year(source)
     case "YEAROFWEEK" => YearOfWeek(source)
-    case "QUARTER" | "QTR" => Quarter(source)
```

Comment:
could you please add a comment about why you are adding this here please.

Reply:
There is more context in DATA-11646 bt will add here too.